antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -16,7 +25,189 @@
 from antarest.study.storage.variantstudy.model.command.icommand import MATCH_SIGNATURE_SEPARATOR, ICommand
 from antarest.study.storage.variantstudy.model.model import CommandDTO
 
-MatrixType = List[List[MatrixData]]
+MatrixType = t.List[t.List[MatrixData]]
+
+ALIAS_OPERATOR_MAP = {
+    BindingConstraintOperator.EQUAL: "eq",
+    BindingConstraintOperator.LESS: "lt",
+    BindingConstraintOperator.GREATER: "gt",
+}
+
+
+class MatrixInputTargetPaths(BaseModel, frozen=True, extra="forbid"):
+    """
+    Model used to store the input and target paths for matrices.
+    """
+
+    matrix_input_paths: t.List[Path] = Field(..., min_items=1, max_items=2)
+    matrix_target_paths: t.List[Path] = Field(..., min_items=1, max_items=2)
+
+
+def _infer_input_and_target_paths(
+    parent_node: FolderNode,
+    binding_constraint_id: str,
+    existing_operator: BindingConstraintOperator,
+    new_operator: BindingConstraintOperator,
+) -> MatrixInputTargetPaths:
+    """
+    Infer the target paths of the matrices to update according to the existing and new operators.
+
+    Args:
+        parent_node: the parent folder node
+        binding_constraint_id: the binding constraint ID
+        existing_operator: the existing operator
+        new_operator: the new operator
+
+    Returns:
+        the matrix input and target paths
+    """
+    # new and existing operators should be different
+    assert existing_operator != new_operator, "Existing and new operators should be different"
+
+    matrix_node_lt = parent_node.get_node([f"{binding_constraint_id}_lt"])
+    assert isinstance(
+        matrix_node_lt, LazyNode
+    ), f"Node type not handled yet: LazyNode expected, got {type(matrix_node_lt)}"
+    matrix_node_eq = parent_node.get_node([f"{binding_constraint_id}_eq"])
+    assert isinstance(
+        matrix_node_eq, LazyNode
+    ), f"Node type not handled yet: LazyNode expected, got {type(matrix_node_eq)}"
+    matrix_node_gt = parent_node.get_node([f"{binding_constraint_id}_gt"])
+    assert isinstance(
+        matrix_node_gt, LazyNode
+    ), f"Node type not handled yet: LazyNode expected, got {type(matrix_node_gt)}"
+
+    is_link_eq = matrix_node_eq.infer_is_link_path()
```

Comment:
In this file we should not have to manipulate paths directly, we should instead manipulate nodes of the tree.

For example to copy lt file to gt file, we would just call:
```python
matrix_node_lt.copy_to(matrix_node_gt) 
# or maybe just a path in the argument ?
```

This way all the link-related complexities will be handled by the nodes themselves which have the right information internally, and hidden from here.